Content.Server/ImmovableRod/ImmovableRodSystem.cs

```diff
@@ -94,12 +97,28 @@ private void OnCollide(EntityUid uid, ImmovableRodComponent component, ref Start
             return;
         }
 
-        // gib em
+        // dont delete/hurt self if polymoprhed into a rod
+        if (TryComp<PolymorphedEntityComponent>(uid, out var polymorphed))
+        {
+            if (polymorphed.Parent == ent)
+                return;
+        }
+
+        // gib or damage em
         if (TryComp<BodyComponent>(ent, out var body))
         {
             component.MobCount++;
-
             _popup.PopupEntity(Loc.GetString("immovable-rod-penetrated-mob", ("rod", uid), ("mob", ent)), uid, PopupType.LargeCaution);
+
+            if (!component.ShouldGib)
+            {
+                if (!TryComp<DamageableComponent>(ent, out var damageable) || component.Damage == null)
+                    return;
```

Comment:
Checking null is significantly faster than trycomp so you should reverse this order.

Reply:
flipped